flink-clients/src/main/java/org/apache/flink/client/deployment/application/UnsuccessfulTerminationException.java

```diff
@@ -32,11 +32,11 @@
  * application with a given {@link ApplicationStatus}.
  */
 @Internal
-public class ApplicationFailureException extends JobExecutionException {
+public class UnsuccessfulTerminationException extends JobExecutionException {
```

Comment:
Isn't this exception more about an unsuccessful execution or completion than the termination? The termination of the cluster will happen as a result of this exception, right?

Reply:
Yes, I can rename it to `UnsuccessfulExecutionException`. And yes, the cluster is going to be shutdown one way or another.